### PR TITLE
SDK-379 [Java SDK] Introduce throttling to http requests

### PR DESCRIPTION
**Pull request by VladimirLevitskiy**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-379](https://incountry.atlassian.net/browse/SDK-379)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: src/main/java/com/incountry/residence/sdk/tools/http/impl/ThrottlingRequestExecutor.java [hunk 1]
```diff
@@ -16,51 +18,97 @@
...
-    private static final Logger LOG = LogManager.getLogger(AbstractHttpRequestCreator.class);
+public abstract class ThrottlingRequestExecutor {
+    private static final Logger LOG = LogManager.getLogger(ThrottlingRequestExecutor.class);
 
     private static final String MSG_ERR_URL = "URL error";
     private static final String MSG_ERR_NULL_BODY = "HTTP request body can't be null";
+    private static final String MSG_WARN_TOO_MANY_REQUESTS = "Storage server error: too many requests. Will retry in {} ms";
+
+    private static final String USER_AGENT = "User-Agent";
+    private static final String USER_AGENT_VALUE = "SDK-Java/" + Version.BUILD_VERSION;
```
Comment (RommeroInCountry) [inline src/main/java/com/incountry/residence/sdk/tools/http/impl/ThrottlingRequestExecutor.java:39]: it's kinda wrong to do headers in this class. I'd leave them be as they are now. Or even moving them further higher in stack. These are just custom headers for the request that should be simply moved further down the pipe. Let's discuss it on the DSU call

Comment (RommeroInCountry) [reply src/main/java/com/incountry/residence/sdk/tools/http/impl/ThrottlingRequestExecutor.java:39]: p.s. USER_AGENT -> USER_AGENT_HEADER_NAME, USER_AGENT_VALUE -> USER_AGENT_HEADER_VALUE

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]
Let's have a call on the proposed changes

**Comment by VladimirLevitskiy** [comment]
> Let's have a call on the proposed changes

done

**Review by RommeroInCountry — APPROVED** [review]

**Review by robert-par — APPROVED** [review]

**Pull request merged**